File: src/domain/repositories/project-repository.ts
import type { Project } from '../entities/project.js';
import type { ProjectId } from '../value-objects/project-id.js';

export interface ProjectRepository {
  findById: (_id: ProjectId) => Promise<Project | null>;
  save: (_project: Project) => Promise<void>;
  delete: (_id: ProjectId) => Promise<void>;
  findAll: () => Promise<Project[]>;
}

File: src/main/kotlin/io/spiralhouse/jcvd/domain/repositories/ProjectRepository.kt
package com.spiralhouse.jcvd.domain.repositories

import com.spiralhouse.jcvd.domain.entities.Project
import com.spiralhouse.jcvd.domain.valueobjects.ProjectId
import com.spiralhouse.jcvd.domain.valueobjects.ProjectStatus

interface ProjectRepository {
    suspend fun findById(id: ProjectId): Project?
    suspend fun findByStatus(status: ProjectStatus): List<Project>
    suspend fun findAll(): List<Project>
    suspend fun save(project: Project)
    suspend fun delete(id: ProjectId)
    suspend fun exists(id: ProjectId): Boolean
}